File: python/CP/HackerRank/insertionsort2.py
def insertion_sort(arr, length):

    for y in range(1, length):
        e = arr[y]
        x = y

        while (x>0) and (arr[x-1] > e):
            arr[x] = arr[x-1]
            x -= 1

        arr[x] = e
        print(*arr)


length = int(input())
arr = list(map(int, input().split()))
insertion_sort(arr, length)

File: Competive Programming/Ruby/HackerRank/insertionsort2.rb
def insertion_sort(arr, len)
    1.upto(len-1) do |x|
        e = arr[x]
        y = x
        
        while (y>0) && (arr[y-1] > e)
            arr[y] = arr[y-1]
            y -= 1
        end
        
        arr[y] = e
        puts arr.join(" ")
    end
end


len = gets.chomp.to_i
arr = gets.chomp.split.map { |x| x.to_i }
insertion_sort(arr, len)
